File: Shared/Java/harrycheung/map/shared/Session.java
//
// Copyright (c) 2015 Harry Cheung
//

package harrycheung.map.shared;

import java.util.ArrayList;
import java.util.List;

public final class Session {

  protected Track track;
  protected double startTime;
  protected double duration = 0;
  protected List<Lap> laps = new ArrayList<Lap>();

  public Session(Track track, double startTime) {
    this.track     = track;
    this.startTime = startTime;
  }

  public void tick(double currentTime) {
    duration = currentTime - startTime;
  }
}

File: Unity/Mobile-App-Performance/Assets/Scripts/Shared/Session.cs
//
// Copyright (c) 2015 Harry Cheung
//

using System;
using  System.Collections.Generic;

namespace Xamarin.Shared
{
	public class Session
	{
		public Track track;
		public double startTime;
		public double duration = 0;
		public List<Lap> laps = new List<Lap>();

		public Session(Track track, double startTime) {
			this.track     = track;
			this.startTime = startTime;
		}
	}
}

